src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/dashboard/fix/data-broker-profiles/welcome-to-premium/page.tsx

```diff
@@ -55,6 +56,10 @@ export default async function WelcomeToPlusPage() {
   // Let’s make sure the users OneRep profile is activated:
   await activateAndOptoutProfile({ profileId, forceActivation: true });
 
+  // Make sure the current state of the stored scan results is being reflected
+  // after we just initiated automatic removal.
+  await refreshStoredScanResults(profileId);
```

Comment:
We've got a bunch of function calls in our code base now where we're not sure whether they're actually needed, which carries a high risk of turning our codebase into an append-only codebase - we're never sure what we can remove.

Could we add a comment here that explains that we're not sure whether this is actually needed, what behaviour we're not seeing without it, and what behaviour we're hoping to see with it? Then in the future, we can at least verify whether those conditions still hold, of if we've found a fix.

Reply:
Good call-out. Added in [295edbf](https://github.com/mozilla/blurts-server/pull/4137/commits/295edbfbdb4ffc609b665dae475d2dae6f6baa70).